pkg/registry/apis/provisioning/jobs.go

```diff
@@ -0,0 +1,99 @@
+package provisioning
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/registry/rest"
+
+	provisioning "github.com/grafana/grafana/pkg/apis/provisioning/v0alpha1"
+	"github.com/grafana/grafana/pkg/registry/apis/provisioning/jobs"
+)
+
+type jobsConnector struct {
+	repoGetter RepoGetter
+	jobs       jobs.Queue
+	historic   jobs.History
+}
+
+func (*jobsConnector) New() runtime.Object {
+	return &provisioning.Job{}
+}
+
+func (*jobsConnector) Destroy() {}
+
+func (*jobsConnector) ProducesMIMETypes(verb string) []string {
+	return []string{"application/json"}
+}
+
+func (c *jobsConnector) ProducesObject(verb string) any {
+	return c.New()
+}
+
+func (*jobsConnector) ConnectMethods() []string {
+	return []string{http.MethodPost, http.MethodGet}
+}
+
+func (*jobsConnector) NewConnectOptions() (runtime.Object, bool, string) {
+	return nil, false, ""
+}
+
+func (c *jobsConnector) Connect(
+	ctx context.Context,
+	name string,
+	opts runtime.Object,
+	responder rest.Responder,
+) (http.Handler, error) {
+	repo, err := c.repoGetter.GetHealthyRepository(ctx, name)
+	if err != nil {
+		return nil, err
+	}
+	cfg := repo.Config()
+
+	return withTimeout(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == http.MethodGet {
+			query := r.URL.Query()
+			if query.Has("job") {
+				job, err := c.historic.GetJob(ctx, name, query.Get("job"))
+				if err != nil {
+					responder.Error(err)
+					return
+				}
+				responder.Object(http.StatusOK, &provisioning.JobList{Items: []provisioning.Job{*job}})
```

Comment:
Why a job list and not only 1 ? 

Can we have 2 endpoints ? 

`/jobs`
`/jobs/<name>`